src/compiler/transformers/declarations.ts

```diff
@@ -214,7 +215,16 @@ import {
 /** @internal */
 export function getDeclarationDiagnostics(host: EmitHost, resolver: EmitResolver, file: SourceFile | undefined): DiagnosticWithLocation[] | undefined {
     const compilerOptions = host.getCompilerOptions();
-    const result = transformNodes(resolver, host, factory, compilerOptions, file ? [file] : filter(host.getSourceFiles(), isSourceFileNotJson), [transformDeclarations], /*allowDtsFiles*/ false);
+    const files = filter(getSourceFilesToEmit(host, file), isSourceFileNotJson);
+    const result = transformNodes(
+        resolver,
+        host,
+        factory,
+        compilerOptions,
+        file ? contains(files, file) ? [file] : emptyArray : files,

```

Comment:
Readability nit, but I have a real hard time trying to read this one; is there another way to put this?

```ts
let files = filter(getSourceFilesToEmit(host, file), isSourceFileNotJson);
if (file) {
   files = contains(files, file) ? [file] : emptyArray;
}
```

Maybe? If not, no worries, I just have problems reading ternaries.

Reply:
Sorry its hard to read for you. I prefer `consts` over let for this. Do you think having it multi line will be more readable

Reply:
It's fine as it is, no worries.